src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/NuGetProjectService.cs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Threading;
+using NuGet.Common;
+using NuGet.VisualStudio.Contracts;
+
+namespace NuGet.VisualStudio.Implementation.Extensibility
+{
+    internal class NuGetProjectService : INuGetProjectService
+    {
+        private readonly IProjectSystemCache _projectSystemCache;
+
+        public NuGetProjectService(IProjectSystemCache projectSystemCache)
+        {
+            _projectSystemCache = projectSystemCache ?? throw new ArgumentNullException(nameof(projectSystemCache));
+        }
+
+        public async Task<GetInstalledPackagesResult> GetInstalledPackagesAsync(string project, CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrEmpty(project))
+            {
+                throw new ArgumentNullException(paramName: nameof(project));
+            }
+
+            await TaskScheduler.Default;
+
+            if (!_projectSystemCache.TryGetNuGetProject(project, out var nuGetProject))
```

Comment:
Any reason you didn't use https://github.com/NuGet/NuGet.Client/blob/f616fda3ca687960ac472fe2678c39f0cbca24b3/src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs#L221 instead?

Reply:
I didn't know about it. Does that count? :)

Looking at the implementation, it is just a wrapper around `IProjectSystemCache.TryGetNuGetProject`, while [duplicating the same input validation](https://github.com/NuGet/NuGet.Client/blob/f616fda3ca687960ac472fe2678c39f0cbca24b3/src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/ProjectSystemCache.cs#L47-L50), so I don't see a benefit of using it, rather than the project system cache directly.

Reply:
It does seem odd that we would even expose the project system cache tbh, as the solutionmanager hooks to solution events to update the cache to make sure it is not stale.  IMO having one place responsible for getting projects instead of having two ways to do the same thing is bad and I would argue in favor of using the solutionmanager.